src/xtopcom/xrpc/eth_rpc/eth_method.cpp

```diff
@@ -31,22 +31,22 @@ void EthMethod::web3_sha3(const Json::Value & js_req, Json::Value & js_rsp) {
 void EthMethod::eth_gasPrice(const Json::Value & js_req, Json::Value & js_rsp) {
     if (!eth::EthErrorCode::check_req(js_req, js_rsp, 0))
         return;
-    //Json::Value value = "0x3b9aca00";
-    // Json::Value value = "0x989680";
     auto base_price = top::gasfee::xgas_estimate::base_price();
 
     std::string baseprice_str = top::to_hex((top::evm_common::h256)base_price);
-    uint32_t i = 0;
-    for (; i < baseprice_str.size() - 1; i++) {
-        if (baseprice_str[i] != '0') {
-            break;
-        }
+    std::string::size_type found = baseprice_str.find_first_not_of("0");
```

Comment:
create a new api, may be called  top::to_hex_prefixed_without_leading0

hex.erase(0, hex.find_first_not_of('0'));